backend/pkg/schema/client.go

```diff
@@ -44,6 +46,46 @@ func newClient(cfg Config) (*Client, error) {
 		client = client.SetAuthToken(cfg.BearerToken)
 	}
 
+	// Configure Client Certificat transport
+	if cfg.TLS.Enabled {
+
+		cert, err := ioutil.ReadFile(cfg.TLS.CertFilepath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read cert file for schema registry client: %w", err)
+		}
+
+		privateKey, err := ioutil.ReadFile(cfg.TLS.KeyFilepath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read key file for schema registry client: %w", err)
+		}
+
+		pemBlock, _ := pem.Decode(privateKey)
+		if pemBlock == nil {
+			return nil, fmt.Errorf("no valid private key found")
+		}
+		if x509.IsEncryptedPEMBlock(pemBlock) {
+			decryptedKey, err := x509.DecryptPEMBlock(pemBlock, []byte(cfg.TLS.Passphrase))
+			if err != nil {
+				return nil, fmt.Errorf("private key is encrypted, but could not decrypt it: %s", err)
+			}
+			// If private key was encrypted we can overwrite the original contents now with the decrypted version
+			privateKey = pem.EncodeToMemory(&pem.Block{Type: pemBlock.Type, Bytes: decryptedKey})
+		}
```

Comment:
These methods are marked as deprecated in Go because the PEM encryption itself is insecure by design, see: https://pkg.go.dev/crypto/x509#IsEncryptedPEMBlock

I think we should not add that functionality if it's due to be removed

Reply:
I took this from https://github.com/cloudhut/kowl/blob/e0413820058c13a9f8ebff41450604b20d6ca52c/backend/pkg/kafka/config_helper.go#L152 :)
My PEM is not encryted so I do not need this

Reply:
I assumed that. Back when I implemented these methods were not deprecated and I was not aware that this should no longer be used. I believe it should not be added now that this is known.